static/app/components/profiling/threadSelector.tsx

```diff
@@ -58,7 +72,21 @@ function ThreadMenuSelector({
         icon: <IconList size="xs" />,
         size: 'xs',
       }}
-      options={options}
+      options={[
+        {
+          // TODO: fix CompactSelect types to better represent usage.
+          // this value has no effect on the selection, it's simply to satisfy the types..
+          // tried modifying the types but it creates a lot of errors
+          value: threadId || 0,
```

Comment:
I don't remember but I think you want to use a different `value` for the 2 nested options or the compact select might get confused..